src/silo_api/command_line_arguments.cpp

```diff
@@ -1,26 +1,66 @@
 #include "silo_api/command_line_arguments.h"
 
+#include <boost/algorithm/string/join.hpp>
+
 using silo_api::CommandLineArguments;
 
+std::string CommandLineArguments::asUnixOptionString(
+   const silo::config::AbstractConfigSource::Option& option
+) {
+   std::vector<std::string> result;
+   for (const std::string& current_string : option.access_path) {
+      std::string current_result;
+      for (const char character : current_string) {
+         if (std::isupper(character)) {
+            current_result += '-';
+            const char char_in_lower_case =
+               static_cast<char>(std::tolower(static_cast<unsigned char>(character)));
+            current_result += char_in_lower_case;
+         } else {
+            current_result += character;
+         }
+      }
+      result.emplace_back(current_result);
+   }
+   return "SILO_" + boost::join(result, "-");
+}
+
 CommandLineArguments::CommandLineArguments(const Poco::Util::AbstractConfiguration& config)
     : config(config) {}
 
 std::string CommandLineArguments::configType() const {
    return "command line argument";
 }
 
-bool CommandLineArguments::hasProperty(const std::string& key) const {
-   return config.hasProperty(key);
+bool CommandLineArguments::hasProperty(const Option& option) const {
+   return config.hasProperty(asUnixOptionString(option));
 }
 
-std::string CommandLineArguments::getString(const std::string& key) const {
-   return config.getString(key);
+std::optional<std::string> CommandLineArguments::getString(const Option& option) const {
+   if (hasProperty(option)) {
+      return config.getString(asUnixOptionString(option));
+   }
+   return std::nullopt;
 }
 
-int32_t CommandLineArguments::getInt32(const std::string& key) const {
-   return config.getInt(key);
+std::optional<int32_t> CommandLineArguments::getInt32(const Option& option) const {
+   if (hasProperty(option)) {
+      try {
+         return config.getInt(asUnixOptionString(option));
+      } catch (Poco::SyntaxException&) {
+         return std::nullopt;
```

Comment:
Same here, I think it would be better if we don't silently ignore this exception.